core/tls-websocket-server/src/certificate_generation.rs

```diff
@@ -67,7 +67,7 @@ pub fn ed25519_self_signed_certificate(
 ) -> WebSocketResult<Certificate> {
 	let mut params = CertificateParams::new(vec![common_name.to_string()]);
 	let now = SystemTime::now().duration_since(UNIX_EPOCH).expect("Error: UNIX_EPOCH");
-	let issue_ts = TzUtc.timestamp(now.as_secs() as i64, 0);
+	let issue_ts = TzUtc.timestamp_opt(now.as_secs() as i64, 0).unwrap();
```

Comment:
If you know some context about the `None` here, I would prefer an `expect`. If it is something that never fails in our case, it is even better, and you can write like: `expect("fails never because of ....; qed)`.

Reply:
This should not fail as long as `now.as_secs()` (aka `secs`) fit into `i64`. The `as` keyword will silently convert it into -1 (e.g. when `secs == u64::MAX`) and that's when it would fail.

Revised in 89e7ac2c.